src/js/media-chrome-dialog.ts

```diff
@@ -1,52 +1,68 @@
-import { globalThis, document } from './utils/server-safe-globals.js';
+import { globalThis } from './utils/server-safe-globals.js';
 import {
   containsComposedNode,
   getActiveElement,
+  getBooleanAttr,
+  namedNodeMapToObject,
+  setBooleanAttr,
 } from './utils/element-utils.js';
 import { InvokeEvent } from './utils/events.js';
 
-const template: HTMLTemplateElement = document.createElement('template');
-template.innerHTML = /*html*/ `
-  <style>
-    :host {
-      font: var(--media-font,
-        var(--media-font-weight, normal)
-        var(--media-font-size, 14px) /
-        var(--media-text-content-height, var(--media-control-height, 24px))
-        var(--media-font-family, helvetica neue, segoe ui, roboto, arial, sans-serif));
-      color: var(--media-text-color, var(--media-primary-color, rgb(238 238 238)));
-      background: var(--media-dialog-background, var(--media-control-background, var(--media-secondary-color, rgb(20 20 30 / .8))));
-      border-radius: var(--media-dialog-border-radius);
-      border: var(--media-dialog-border, none);
-      display: var(--media-dialog-display, inline-flex);
-      transition: var(--media-dialog-transition-in,
-        visibility 0s,
-        opacity .2s ease-out,
-        transform .15s ease-out
-      ) !important;
-      ${/* ^^Prevent transition override by media-container */ ''}
-      visibility: var(--media-dialog-visibility, visible);
-      opacity: var(--media-dialog-opacity, 1);
-      transform: var(--media-dialog-transform-in, translateY(0) scale(1));
-    }
+function getTemplateHTML(_attrs: Record<string, string>) {
```

Comment:
suggestion: add a code comment documenting that this is intended to be used as a mixin static method on a class that also has a static method called `getSlotTemplateHTML()` somewhere in the class inheritance chain. I suspect the relationships between these two when subclasses are involved will not be super intuitive (definitely required some chewing on my part for the different permutations).